tests/unit/expire.tcl

```diff
@@ -601,4 +601,117 @@ start_server {tags {"expire"}} {
            {del foo}
         }
     } {} {needs:repl}
+
+    test {EXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {EXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200000 NX] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {PEXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r PEXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
```

Comment:
i'm actually not sure if there's high value to test the P variants at all (considering they use common code).
instead (or in addition to), i'd maybe we can add one test that uses NX/XX/GT/LT on a key that's totally missing?
also kinda pointless since this code is unreachable today when the key is missing, but maybe it'll expose an issue when the code is changed one day.